ui/src/services/PodUtils.js

```diff
@@ -8,16 +9,21 @@ export const getPodsListData = (nodeName, pods) => {
     podsList?.map((pod) => {
       const age = fromMilliSectoAge(new Date() - pod.startTime);
 
-      const numContainer = pod?.containerStatuses?.length;
-      const containerReady =
-        pod?.containerStatuses?.filter((pCS) => pCS.ready === true) ?? [];
+      const numContainer =
+        pod.status === STATUS_RUNNING ? pod?.containerStatuses?.length : 0;
```

Comment:
There's a small chance that you get `undefined` out of this:

```suggestion
        pod.status === STATUS_RUNNING
        ? (pod?.containerStatuses?.length ?? 0)
        : 0;
```

Reply:
Side-note: you can also determine the number of expected containers by looking at `pod.spec.containers.length`

Reply:
Indeed!!! I may get `undefined` for `numContainer`. So I made the following changes.

```javascript
const numContainer = pod?.containerStatuses?.length ?? 0;
```

I didn't use the `pod.spec.containers.length` because I'll need to retrieve one more field and store it in redux store. Since I we already have `containerStatuses` and the length should be the same as `pod.spec.containers.length`